src/util.rs

```diff
@@ -267,8 +267,17 @@ pub mod git {
                 "remote authentication required but none available",
             ))
         });
+
+        // Creating a proxy option to, eventually, address the usage of Sheldon
+        // behind a http proxy. Using the [auto()] function to try to
+        // auto-detect the proxy from the git configuration.
```

Comment:
I do appreciate comments but this feels overly verbose. Perhaps we can just make this.

```suggestion
        // Try to auto-detect the proxy from the git configuration so that
        // Sheldon can be used behind a proxy.
```